runtime/auth/signing/aws-signing-tests/jvm/src/aws/smithy/kotlin/runtime/auth/signing/awssigning/tests/SigningSuiteTestBaseJVM.kt

```diff
@@ -0,0 +1,361 @@
+import kotlin.io.path.name
+import kotlin.io.path.readText
+import kotlin.streams.toList
+import kotlin.test.*
+import kotlin.time.Duration.Companion.seconds
+
+private const val DEFAULT_SIGNING_ISO_DATE = "2015-08-30T12:36:00Z"
+
+private val defaultTestCredentialsProvider =
+    StaticCredentialsProvider(Credentials("AKIDEXAMPLE", "wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY"))
+
+private val defaultTestSigningConfig = AwsSigningConfig.Builder().apply {
+    algorithm = AwsSigningAlgorithm.SIGV4
+    credentialsProvider = defaultTestCredentialsProvider
+    signingDate = Instant.fromIso8601(DEFAULT_SIGNING_ISO_DATE)
+    region = "us-east-1"
+    service = "service"
+    useDoubleUriEncode = true
+    normalizeUriPath = true
+}
+
+data class Sigv4TestSuiteTest(
+    val path: Path,
+    val request: HttpRequestBuilder,
+    val signedRequest: HttpRequestBuilder,
+    val config: AwsSigningConfig = defaultTestSigningConfig.build()
+)
+
+private val testSuitePath: Path by lazy {
+    val cl = ClassLoader.getSystemClassLoader()
+    val url = cl.getResource("aws-signing-test-suite") ?: error("failed to load sigv4 test suite resource")
+    val uri = url.toURI()
+    FileSystems.newFileSystem(uri, mapOf<String, Any>()).getPath("/aws-signing-test-suite/v4")
+}
+
+// FIXME - move to common test (will require ability to access test resources in a KMP compatible way)
+actual abstract class SigningSuiteTestBase : HasSigner {
+    private val testDirPaths: List<Path> by lazy {
+        Files
+            .walk(testSuitePath)
+            .toList()
+            .filter { !it.isDirectory() && it.name == "request.txt" }
+            .filterNot { it.parent.name in disabledTests }
+            .map { it.parent }
+    }
+
+    protected open val disabledTests = setOf<String>()
+
+    @Test
+    fun testParseRequest() {
+        // sanity test that we are converting requests from file correctly
+        val noBodyTest = testSuitePath.resolve("post-vanilla")
+        val actual = getSignedRequest(noBodyTest)
+
+        assertEquals(3, actual.headers.names().size)
+        assertIs<HttpBody.Empty>(actual.body)
+        assertEquals("example.amazonaws.com", actual.headers["Host"])
+        assertEquals("20150830T123600Z", actual.headers["X-Amz-Date"])
+        assertEquals(
+            "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, Signature=5da7c1a2acd57cee7505fc6676e4e544621c30862966e37dddb68e92efbe5d6b",
+            actual.headers["Authorization"]
+        )
+    }
+
+    @Test
+    fun testSigv4TestSuiteHeaders() {
+        val tests = getTests(AwsSignatureType.HTTP_REQUEST_VIA_HEADERS)
+        testSigv4Middleware(tests)
+    }
+
+    @Test
+    fun testSigv4TestSuiteQuery() {
+        val tests = getTests(AwsSignatureType.HTTP_REQUEST_VIA_QUERY_PARAMS)
+        testSigv4Middleware(tests)
+    }
+
+    private fun getTests(signatureType: AwsSignatureType): List<Sigv4TestSuiteTest> {
```

Comment:
Actually since these run in JVM, we can switch to parameterized tests pretty easily. They won't show names on the CLI output because of [gradle#5975](https://github.com/gradle/gradle/issues/5975) but they run as separate tests and the HTML output shows them properly.

Reply:
The main driving factor was in setting breakpoints or running a single test in isolation I'm not sure parameterized tests help there does it?